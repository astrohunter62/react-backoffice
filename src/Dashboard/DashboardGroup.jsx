import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from 'material-ui/styles'

import Grid from 'material-ui/Grid'
import Typography from 'material-ui/Typography'

import DashboardCard from './DashboardCard'

const styles = theme => ({
  headline: {
    marginTop: theme.spacing.unit * 3,
    marginBottom: theme.spacing.unit * 2,
  },
})

const DashboardGroup = ({
  title,
  cards,
  onClick,
  classes,
}) => (
  <div>
    <div className={classes.headline}>
      <Typography variant="headline">
        {title}
      </Typography>
    </div>

    <Grid container spacing={16}>

      {cards ? cards.map(card => (
        <Grid item xs={12} sm={4} key={`group-card-${card.id}`}>
          <DashboardCard
            title={card.title}
            description={card.description}
            handleClick={() => onClick(card.link)}
            icon={card.icon}
            disabled={card.disabled}
          />
        </Grid>
      )) : null}

    </Grid>
  </div>
)

DashboardGroup.propTypes = {
  title: PropTypes.string.isRequired,
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(DashboardGroup)
