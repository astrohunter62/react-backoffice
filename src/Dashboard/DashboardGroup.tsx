import React from "react";
import { Grid, Typography, withStyles } from "@material-ui/core";
import DashboardCard from "./DashboardCard";
const styles = (theme: any) => ({
  headline: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(2),
  },
});

type DashboardGroupProps = {
  title: string;
  cards: Record<string, any>[];
  onClick: (...args: any[]) => any;
  classes: {
    [key: string]: string;
  };
};
const DashboardGroup: React.SFC<DashboardGroupProps> = ({
  title,
  cards,
  onClick,
  classes,
}) => (
  <div>
    <div className={classes.headline}>
      <Typography variant="h5">{title}</Typography>
    </div>

    <Grid container spacing={2}>
      {cards
        ? cards.map((card) => (
            <Grid item xs={12} sm={4} key={`group-card-${card.id}`}>
              <DashboardCard
                title={card.title}
                description={card.description}
                handleClick={() => onClick(card.link)}
                icon={card.icon}
                isDisabled={card.isDisabled}
              />
            </Grid>
          ))
        : null}
    </Grid>
  </div>
);

export default withStyles(styles)(DashboardGroup);
