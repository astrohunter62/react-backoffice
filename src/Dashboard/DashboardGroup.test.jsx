import React from 'react'
import { shallow } from 'enzyme'

import DashboardGroup from './DashboardGroup'

it('renders correctly', () => {
  const tree = shallow(<DashboardGroup
    title="Title"
    cards={[{
      title: 'title',
      description: 'desc',
    }]}
    onClick={() => {}}
  />)

  expect(tree).toMatchSnapshot()
})
