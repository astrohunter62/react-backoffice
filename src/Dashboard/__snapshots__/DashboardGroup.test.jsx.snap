// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="DashboardGroup-headline-1"
  >
    <h1
      className="MuiTypography-root-2 MuiTypography-headline-7"
    >
      Title
    </h1>
  </div>
  <div
    className="MuiGrid-container-26 MuiGrid-spacing-xs-16-48"
  >
    <div
      className="MuiGrid-item-27 MuiGrid-grid-xs-12-63 MuiGrid-grid-sm-4-68"
    >
      <div
        className="MuiPaper-root-123 MuiPaper-elevation2-127 MuiPaper-rounded-124 MuiCard-root-122 DashboardCard-root-116"
        onClick={[Function]}
      >
        <div
          className="MuiCardContent-root-150 DashboardCard-content-117"
        >
          <h2
            className="MuiTypography-root-2 MuiTypography-headline-7 DashboardCard-title-119"
          >
            title
          </h2>
          <p
            className="MuiTypography-root-2 MuiTypography-body1-11"
          >
            desc
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
